libshaderc_spvc/src/spvc_private.cc

```diff
@@ -321,10 +325,9 @@ shaderc_spvc_compilation_result_t generate_hlsl_shader(
     return result;
   }
 
-  compiler.reset(new spirv_cross::CompilerHLSL(ir));
+  compiler = new (std::nothrow) spirv_cross::CompilerHLSL(*ir);
```

Comment:
I read down into spirv_cross::CompilerHLSL(const spirv_cross::ParsedIR&) and it looks like it makes a copy of the IR object.  You could use std::move here to transfer ownership of the contents of the ir object over to the CompilerHLSL. That might save some CPU time?? 

e.g.
```compiler = new (std::nothrow) spirv_cross::CompilerHLSL(std::move(*ir));```

Reply:
This is an optional change. It's slightly tricky, and should be commented with why it's safe to do.